File: 0045-jump-game-ii/0045-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int jump = 0;
        int near = 0;
        int far = 0;

        while (far < nums.length - 1) {
            int maxDistance = 0;
            for (int i = near; i <= far; i++) {
                maxDistance = Math.max(maxDistance, i + nums[i]);
            }

            near = far + 1;
            far = maxDistance;
            jump++;
        }

        return jump;
    }
}

File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int currEnd = 0; // 表示当前可到达最远距离
        int nextEnd = 0; // 表示下一跳可到达最远距离
        int jump = 0;

        for(int i = 0 ; i < nums.size() - 1 ; i++){
            nextEnd = max(nextEnd, i + nums[i]);

            if(i == currEnd) { // 表示抵达当前最远位置
                jump++;
                currEnd = nextEnd; // 更新当前可到达最远位置

                if(currEnd >= nums.size() - 1) break;
            }
        }
        return jump;
    }
};